### PR TITLE
Add login endpoint

### PR DESCRIPTION
**Pull request by mikeatkins**
doesn't really do anything other than accept post request with joi validation on username and password.. and tests go go along with it.

@ohall

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -10,6 +11,20 @@ exports.register = (server, options, next) => {
...
+    path: '/login',
+    config: {
+      validate: {
+        payload: joi.object().keys({
+          username: joi.string().min(3).max(32).required(),
+          password: joi.string().min(8).max(64).required()
+        })
+      }
+    },
+    handler: require(`${controllerDir}/login`).login
```
Comment (ohall) [inline lib/index.js]: What's this require syntax?  I haven't seen this before.

File: eslintrc.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     }
   },
   "rules": {
-    "semi": 2
+    "semi": 2,
+    "no-mixed-spaces-and-tabs": 2
```
Comment (ohall) [inline eslintrc.json]: Spaces, right?  RIGHT?  😆

File: lib/index.js (hunk 1 above)
Comment (mikeatkins) [inline lib/index.js]: So I may be attempting to be overly clever here. There are a few things slightly unusual about this, and I'm not sure which you are asking about. So:

1. Instead of putting the require at the top of the file, I put it inline *inside a function definition* (in this case the register function). I should probably change this in this case, although I'm a bit undecided. There is a compelling case for putting it inline in some cases, which has to do with how node deals with loading modules and circular dependencies. There are a bunch of log posts about how this can lead to unexpected problems (google "node circular dependency"). Generally there are two solutions, one involving doing just what I did here, and the other is to use dependency injection. Having said that, in this case, it is probably overkill and makes the code a little less readable.

2. I used a variable inside a template string to specify the path to the module. This is common practice in some of the other applications I have worked on. Typically each module defines a variable representing the root path to all local modules, and then all the requires use a template string similar to the above (as opposed to lots of path like ./lib/, etc). In this case I only needed a path to the controller directory, but probably this will evolve and be replaced with just a path to the root, and then more detailed path info in require statements. 
I.e.,
```javascript
const LIBROOT = '../lib'
```
and then later
```
require(`${LIBROOT}/controllers/login`)
```
3. I specified the function I needed chained onto the require. This is just shorthand. Had I done something like this at the top of the file:
```javascript
const loginLib = require(`${controllerDir}/login`);
```
then I would have used it here as:
```javascript
handler: loginLib.login
```
So, basically it is just a reference to the login function defined on the object exported in loginLib.

**Comment by mikeatkins** [comment]
I'm going to merge now, but probably make some changes on the `require` statements in the future

File: lib/index.js (hunk 1 above)
Comment (ohall) [reply lib/index.js:25]: This is a great explanation, thanks!  I had not used template strings in node before and this is a great use case.
